ndcube/ndcube.py

```diff
@@ -587,26 +612,57 @@ def _crop(self, lower_corner, upper_corner, wcs, crop_by_values):
                     lower_corner[i] = interval[0]
                 if upper_is_none:
                     upper_corner[i] = interval[-1]
-        # Convert lower and upper corner coords to array indices.
-        lower_indices = world_to_array_index(*lower_corner)
-        upper_indices = world_to_array_index(*upper_corner)
-        # Ensure return type is tuple of lists, even if only one axis returned.
-        if not isinstance(lower_indices, tuple):
-            lower_indices = (lower_indices,)
-            upper_indices = (upper_indices,)
-        # Construct item which which to slice NDCube.
-        item = []
-        for i, (lower, upper) in enumerate(zip(lower_indices, upper_indices)):
-            # If upper limit index less than zero,
-            # then interval does not overlap with cube range. Raise error.
-            if upper < 0:
-                raise IndexError("Input real world interval beyond range of NDCube. "
-                                 f"Axis: {i}; Axis length: {self.dimensions[i]}; "
-                                 f"Derived array indices: {(lower, upper)}; ")
-            # Construct slice for this axis and append to item.
-            # Increment upper idex by 1 to ensure the upper world coord
-            # is included in sliced cube.
-            item.append(slice(max(0, lower), upper + 1))
+
+        return lower_corner, upper_corner
+
+    def _bounding_box_to_points(self, lower_corner_values, upper_corner_values, wcs):
+        """
+        Convert two corners of a bounding box to the points of all corners.
+        """
+        return tuple(itertools.product(*zip(lower_corner_values, upper_corner_values)))
+
+    def _crop_from_points(self, *world_points_values, wcs):
+        """
+        Crop the cube such that all the specified world points fall within it.
```

Comment:
```suggestion
        Crop to the minimum cube in pixel-space that contains all the specified world points.
```

The aim of this suggestion is to clarify that:
1. The new cube is defined in pixel-space
2. It is the minimum region that contains all the points, given the constraint of 1.